creator-node/src/components/replicaSet/URSMRegistrationComponentService.js

```diff
@@ -173,6 +150,71 @@ const respondToURSMRequestForSignature = async ({ libs: audiusLibs, nodeConfig }
   }
 }
 
+const getSPInfo = async ({ audiusLibs, spID, reqTimestamp, reqSignature }) => {
+  if (!spID || !reqTimestamp || !reqSignature) {
+    throw new ErrorBadRequest('Must provide all required query parameters: spID, timestamp, signature')
+  }
+
+  spID = parseInt(spID)
+
+  /**
+     * Fetch node info from L1 ServiceProviderFactory for spID
+     */
+  const spRecordFromSPFactory = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceEndpointInfo(
+    'content-node',
+    spID
+  )
+  let {
+    owner: ownerWalletFromSPFactory,
+    delegateOwnerWallet: delegateOwnerWalletFromSPFactory,
+    endpoint: nodeEndpointFromSPFactory
+  } = spRecordFromSPFactory
+  delegateOwnerWalletFromSPFactory = delegateOwnerWalletFromSPFactory.toLowerCase()
+
+  return {
+    ownerWalletFromSPFactory,
+    delegateOwnerWalletFromSPFactory,
+    nodeEndpointFromSPFactory,
+    parsedSpID: spID
+  }
+}
+
+// Wrapper fn to perform basic validation that the requester is a valid SP and that the request
+// came from the SP itself
+const verifyRequesterIsValidSP = async ({
+  audiusLibs,
+  spID,
+  reqTimestamp,
+  reqSignature,
+  ownerWalletFromSPFactory,
+  delegateOwnerWalletFromSPFactory,
+  nodeEndpointFromSPFactory
+}) => {
+  /**
```

Comment:
we should check for existence of inputs here. when it was all in one function it was implicitly done